src/main/java/com/celements/filebase/AttachmentService.java

```diff
@@ -74,7 +75,26 @@
   private XWikiContext getContext() {
     return (XWikiContext)execution.getContext().getProperty("xwikicontext");
   }
-
+  
+  @Override
+  public XWikiAttachment addAttachment(XWikiDocument doc, byte[] data, String filename, 
+      String username, String comment) throws AttachmentToBigException,
+        AddingAttachmentContentFailedException, DocumentSaveException {
+    ByteArrayInputStream in = null;
+    try {
+      in = new ByteArrayInputStream(data);
+      return addAttachment(doc, in, filename, username, comment);
+    } finally {
+      if(in != null) {
```

Comment:
to improve readability: please use guavas Closables.closeQuietly(in); instead
http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/io/Closeables.html#closeQuietly(java.io.InputStream)